cmd/kube-scheduler/app/server.go

```diff
@@ -295,6 +307,9 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 			OnStoppedLeading: func() {
 				select {
 				case <-ctx.Done():
+					if gracefulShutdownSecureServer != nil {
+						gracefulShutdownSecureServer()
```

Comment:
Why is gracefulShutdownSecureServer not be called at any termination? Why do we call it only when ctx is done and not when some error is returned etc?